src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/AbstractHandlerProvider.cs

```diff
@@ -10,6 +10,7 @@ namespace Microsoft.CommonLanguageServerProtocol.Framework;
 /// <summary>
 /// Manages handler discovery and distribution.
 /// </summary>
+[System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0016:Add public types and members to the declared API", Justification = "Temporarily required for compat - https://github.com/dotnet/roslyn/issues/72251")]
```

Comment:
Now that the source is included in MS.CA.LanguageServer.Protocol we get public API warnings.  However rather than adding these to the public API, my plan is to convert these to internal APIs (consumers of MS.CA.LanguageServer.Protocol should not be using these types).

However, changing the APIs to internal would be a breaking change for the current VS consumers (they don't have IVTs).  Instead I'll be doing a followup PR once everyone has switched over to the source package to make these internal.  So temporarily suppressing the warnings.